src/dftbp/dftbplus/mainio.F90

```diff
@@ -4129,6 +4173,24 @@ subroutine writeMdOut2(fd, isPeriodic, printForces, hasStress, withBarostat, isL
       call printEnergies(dftbEnergy, electronicSolver, deltaDftb, fd)
     end if
 
+    if (allocated(quadrupoleMoment)) then
+      write(fd, "(A)") ' Traceless Quadrupole moment in au'
+      write(fd, "(A, F14.8, A, F14.8, A, F14.8)") ' XX', quadrupoleMoment(1,1), ' YY',&
+          & quadrupoleMoment(2,2), ' ZZ', quadrupoleMoment(3,3)
+      write(fd, "(A, F14.8, A, F14.8, A, F14.8)") ' XY', quadrupoleMoment(1,2), ' XZ',&
+          & quadrupoleMoment(1,3), ' YZ', quadrupoleMoment(2,3)
+      write(fd, "(A)") ' Traceless Quadrupole moment in Debye*Ang'
+      write(fd, "(A, F14.8, A, F14.8, A, F14.8)") ' XX',&
+          & quadrupoleMoment(1,1) * au__Debye * Bohr__AA, ' YY',&
+          & quadrupoleMoment(2,2) * au__Debye * Bohr__AA, ' ZZ',&
+          & quadrupoleMoment(3,3) * au__Debye * Bohr__AA
+      write(fd, "(A, F14.8, A, F14.8, A, F14.8)") ' XY',&
+          & quadrupoleMoment(1,2) * au__Debye * Bohr__AA, ' XZ',&
+          & quadrupoleMoment(1,3) * au__Debye * Bohr__AA, ' YZ',&
+          & quadrupoleMoment(2,3) * au__Debye * Bohr__AA
+      write(fd, *)
+    end if
```

Comment:
There seems to be two copies of the same output here. How about making this a procedure and reusing it?